### PR TITLE
.Net8 and integration tests

### PR DESCRIPTION
**Pull request by TTA777**
- Upgraded to .Net8
- Added a new project containing builders for our entities, which should ease the writing of tests
- Added a shared setup for our integration tests
- Added a few integration tests to demonstrate the new setup

### CODE REVIEW

**Comment by TTA777** [comment]
Notes from meeting today:
- Needs to add seed to autofaker, to ensure test will be deterministic
- Update docker/ infra code to account for dotnet 7 instead
- Updating the current unit tests with the builders is outside the scope of this PR
- Long term, we want to investigate the possibility of adding a strongly typed, generated http client to the integration tests as well

File: coffeecard/CoffeeCard.Library/CoffeeCard.Library.csproj [hunk 1]
```diff
@@ -1,46 +1,39 @@
...
-      <Compile Remove="Migrations\20220519210629_RestApiV2Changes.cs" />
-      <Compile Remove="Migrations\20220519210629_RestApiV2Changes.Designer.cs" />
-      <Compile Remove="Migrations\20191012125617_Initial - test.cs" />
-      <Compile Remove="Migrations\20191012125617_Initial - test.Designer.cs" />
-    </ItemGroup>
-
-</Project>
+  <PropertyGroup>
+    <TargetFramework>net7.0</TargetFramework>
+    <LangVersion>8</LangVersion>
```
Comment (duckth) [inline coffeecard/CoffeeCard.Library/CoffeeCard.Library.csproj:4]:
```suggestion
    <LangVersion>default</LangVersion>
```

File: coffeecard/CoffeeCard.MobilePay.GenerateApi/CoffeeCard.MobilePay.GenerateApi.csproj [hunk 2]
```diff
@@ -1,15 +1,12 @@
...
-      <PackageReference Include="NSwag.CodeGeneration.CSharp" Version="13.17.0" />
-      <PackageReference Include="NSwag.Core" Version="13.17.0" />
-    </ItemGroup>
-
-</Project>
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>net7.0</TargetFramework>
+    <Nullable>enable</Nullable>
+    <LangVersion>8</LangVersion>
```
Comment (duckth) [inline coffeecard/CoffeeCard.MobilePay.GenerateApi/CoffeeCard.MobilePay.GenerateApi.csproj:6]:
```suggestion
    <LangVersion>default</LangVersion>
```

File: coffeecard/CoffeeCard.Tests.Common/Builders/StatisticBuilder.cs [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+namespace CoffeeCard.Tests.Common.Builders
+{
+    [BuilderFor(typeof(Statistic))]
+    public partial class StatisticBuilder
+    {
+        public static StatisticBuilder Simple()
+        {
+            var user = UserBuilder.Simple().Build();
+            return new StatisticBuilder()
+                .WithUser(user); ;
```
Comment (duckth) [inline coffeecard/CoffeeCard.Tests.Common/Builders/StatisticBuilder.cs:12]: An extra semicolon..? Or weird formatting?

File: coffeecard/CoffeeCard.Tests.Integration/CoffeeCard.Tests.Integration.csproj [hunk 4]
```diff
@@ -1,6 +1,6 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk.Web">
   <PropertyGroup>
-    <TargetFramework>net6.0</TargetFramework>
+    <TargetFramework>net7.0</TargetFramework>
     <IsPackable>false</IsPackable>
     <LangVersion>8</LangVersion>
```
Comment (duckth) [inline coffeecard/CoffeeCard.Tests.Integration/CoffeeCard.Tests.Integration.csproj:5]:
```suggestion
    <LangVersion>default</LangVersion>
```

File: coffeecard/CoffeeCard.Tests.Unit/CoffeeCard.Tests.Unit.csproj [hunk 5]
```diff
@@ -1,6 +1,6 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFramework>net6.0</TargetFramework>
+    <TargetFramework>net7.0</TargetFramework>
     <IsPackable>false</IsPackable>
     <LangVersion>8</LangVersion>
```
Comment (duckth) [inline coffeecard/CoffeeCard.Tests.Unit/CoffeeCard.Tests.Unit.csproj:5]:
```suggestion
    <LangVersion>default</LangVersion>
```

File: coffeecard/CoffeeCard.WebApi/CoffeeCard.WebApi.csproj [hunk 6]
```diff
@@ -1,6 +1,6 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk.Web">
   <PropertyGroup>
-    <TargetFramework>net6.0</TargetFramework>
+    <TargetFramework>net7.0</TargetFramework>
     <AspNetCoreHostingModel>InProcess</AspNetCoreHostingModel>
     <LangVersion>8</LangVersion>
```
Comment (duckth) [inline coffeecard/CoffeeCard.WebApi/CoffeeCard.WebApi.csproj:5]:
```suggestion
    <LangVersion>default</LangVersion>
```

**Review by duckth — CHANGES_REQUESTED** [review]
Remember to update DOTNET_VERSION constant in core-build.yml

**Review by duckth — APPROVED** [review]

**Pull request merged**